x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/extractor/esql/EsqlDataExtractorFactory.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.datafeed.extractor.esql;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.xpack.core.ml.datafeed.DatafeedConfig;
+import org.elasticsearch.xpack.ml.datafeed.extractor.DataExtractor;
+import org.elasticsearch.xpack.ml.datafeed.extractor.DataExtractorFactory;
+
+import java.util.Objects;
+
+public class EsqlDataExtractorFactory implements DataExtractorFactory {
+
+    private final Client client;
+    private final DatafeedConfig datafeed;
+    private final String timeField;
+
+    private EsqlDataExtractorFactory(Client client, DatafeedConfig datafeed, String timeField) {
+        this.client = Objects.requireNonNull(client);
+        this.datafeed = Objects.requireNonNull(datafeed);
+        this.timeField = timeField;
```

Comment:
To rephrase the big comment on `EsqlDataExtractor.java`, there's an assumption that this `timeField` is the field that comes _out_ of the pipeline that the end user provided. But, for efficiency, we might need the time field that goes _in_.